### PR TITLE
Compatibility update with GNFW

### PR DESCRIPTION
**Pull request by williyamshoe**
Updated the `KinConstraintsComposite` class so that it is compatible with [Lenstronomy #650](https://github.com/lenstronomy/lenstronomy/pull/650).  `KinConstraintsComposite` now accepts `alpha_Rs_array` as input, or alternatively `kappa_s_array` or `rho0_array`.

### CODE REVIEW

File: hierarc/LensPosterior/kin_constraints_composite.py [hunk 1]
```diff
@@ -172,10 +182,10 @@ def __init__(
         self._gamma_in_prior_std = gamma_in_prior_std
 
         if not is_m2l_population_level and not self._check_arrays(
-            self._kappa_s_array, log_m2l_array
+            self._alpha_Rs_or_kappa_s_array, log_m2l_array
```
Comment (ajshajib) [inline hierarc/LensPosterior/kin_constraints_composite.py:185]:
```suggestion
            self._halo_normalization_array, log_m2l_array
```

File: hierarc/LensPosterior/kin_constraints_composite.py [hunk 2]
```diff
@@ -210,22 +220,24 @@ def draw_lens(self, no_error=False):
...
                     self._r_eff,
                     1,
                 )
 
-        random_index = np.random.randint(low=0, high=len(self._kappa_s_array))
-        kappa_s_draw = self._kappa_s_array[random_index]
+        random_index = np.random.randint(
+            low=0, high=len(self._alpha_Rs_or_kappa_s_array)
+        )
+        alpha_Rs_or_kappa_s_draw = self._alpha_Rs_or_kappa_s_array[random_index]
```
Comment (ajshajib) [inline hierarc/LensPosterior/kin_constraints_composite.py:240]:
```suggestion
        halo_normalization_draw = self._alpha_Rs_or_kappa_s_array[random_index]
```

**Review by ajshajib — COMMENTED** [review]
Thanks a lot, @williyamshoe. I am requesting a minor variable name change for simplicity.

File: hierarc/LensPosterior/kin_constraints_composite.py [hunk 3]
```diff
@@ -152,16 +155,23 @@ def __init__(
             log_m2l_scaling=log_m2l_scaling,
         )
 
-        if self._check_arrays(kappa_s_array, r_s_angle_array):
-            self._kappa_s_array = kappa_s_array
+        if self._check_arrays(alpha_Rs_array, r_s_angle_array):
+            self._halo_normalization_array = alpha_Rs_array
+            self._alpha_Rs_or_kappa_s_bool = True
```
Comment (ajshajib) [inline hierarc/LensPosterior/kin_constraints_composite.py:160]: Sorry that I didn't explicitly say about this the first time. `self._alpha_Rs_or_kappa_s_bool` is not particularly a clear naming for what it means when it's `True`. I suggest, `self._is_normalization_alpha_Rs` would be clearer.

**Review by ajshajib — APPROVED** [review]
I have one more small suggestion, after that it's good to be merged. Thanks again!

File: hierarc/LensPosterior/kin_constraints_composite.py (hunk 3 above)
Comment (williyamshoe) [reply hierarc/LensPosterior/kin_constraints_composite.py]: Done!  Thanks for reviewing the changes!

**Pull request merged**